src/hooks/useBalance.ts

```diff
@@ -9,67 +9,69 @@ import useCurrentNetwork from './useCurrentNetwork';
 import useWalletService from './useWalletService';
 
 function useBalance(from: Token | undefined | null): [BigNumber | undefined, boolean, string?] {
-  const [isLoading, setIsLoading] = React.useState(false);
   const walletService = useWalletService();
-  const [result, setResult] = React.useState<BigNumber | undefined>(undefined);
-  const [error, setError] = React.useState<string | undefined>(undefined);
+  const [state, setState] = React.useState<{ isLoading: boolean; result?: BigNumber; error?: string }>({
+    isLoading: false,
+    result: undefined,
+    error: undefined,
+  });
+
   const hasPendingTransactions = useHasPendingTransactions();
   const prevFrom = usePrevious(from);
   const prevPendingTrans = usePrevious(hasPendingTransactions);
-  const account = usePrevious(walletService.getAccount());
+  const prevAccount = usePrevious(walletService.getAccount());
+  const account = walletService.getAccount();
   const currentNetwork = useCurrentNetwork();
   const blockNumber = useBlockNumber(currentNetwork.chainId);
   const prevBlockNumber = usePrevious(blockNumber);
-  const prevResult = usePrevious(result, false);
+  const prevResult = usePrevious(state.result, false);
 
   React.useEffect(() => {
     async function callPromise() {
       if (from) {
         try {
           const promiseResult = await walletService.getBalance(from.address);
-          setResult(promiseResult);
-          setError(undefined);
+          setState({ isLoading: false, result: promiseResult, error: undefined });
         } catch (e) {
-          setError(e);
+          setState((prevState) => ({ ...prevState, error: e, isLoading: false }));
```

Comment:
we probably don't want to spread the previous state, since we want to set result as undefined if it has failed